### ISSUE TITLE
Super-class Labels of tieredImageNet

### ISSUE THREAD

**Issue by shelhamer** [renmengye/few-shot-ssl-public#8]
The general labels in each split seem to have an extra, corrupted(?) value that many data points take on. For instance in the training set there are 20 labels {0, ..., 19} but there is a label value 20 used by 341546 data points.

Should all of these data points be excluded, or is there a way to generate correct labels for these?

Apologies if I am confused in my understanding of your dataset. Thank you for working on a hierarchical few-shot dataset.

Here is some illustrative code from my exploration of the data:

```
>>> train['label_general_str']
['garment',
 'musical instrument, instrument',
 'restraint, constraint',
 'feline, felid',
 'instrument',
 'hound, hound dog',
 'electronic equipment',
 'passerine, passeriform bird',
 'ungulate, hoofed mammal',
 'aquatic bird',
 'snake, serpent, ophidian',
 'primate',
 'protective covering, protective cover, protect',
 'terrier',
 'saurian',
 'building, edifice',
 'establishment',
 'tool',
 'craft',
 'game equipment']
>>> len(train['label_general_str'])
20
>>> train['label_general'].max()  # should be 19
20
>>> uniq, count = np.unique(train['label_general'], return_counts=True)
>>> count
array([  1300,   1300,   1216,   1300,   1300,   1300,   1300,   1300,
         1300,   2600,   1300,   2449,   2600,  11700,   2590,  10258,
        13587,  13000,  24158,  11291, 341546])  # many invalid points with last value
```

**Comment by LeslieChen233** [comment]
man, can you run his codes successfully?

**Comment by eleniTriantafillou** [comment]
Thank you for raising this issue! We have updated our code for this and re-generated the dataset's "general" labels (the link provided in the README now points to the new version). Please note that the "general" labels were not used in our work (so this did not affect our results in any way) but we have addressed this in order to encourage future work to leverage this hierarchy.

**Comment by shelhamer** [comment]
Closing as solved by #9 and 1e9fe081be9560596fb3e5c4de01f0162cd65d3b. Thanks!
